src/heal.js

```diff
@@ -13,16 +13,18 @@ heal.config(['$provide', function ($provide) {
             return func().catch(error => {
                 // Check whether session has expired
                 if (error && error.status === 419) {
-                    $log.info('Invalid session, attempting to re-establish');
+                  $log.info('Invalid session, attempting to re-establish');
 
                   // Attempt to re-establish the session once before
                   return reEstablishSession().then(data => {
                         $log.info('Session re-established');
+                        $rootScope.$emit('pandular:re-establishment:success', data);
 
                         // Try again now we have a fresh session
                         return func();
                     }, sessionError => {
                         $log.error('Could not re-establish session: ' + sessionError);
+                        $rootScope.$emit('pandular:re-establishment:fail', sessionError);
```

Comment:
After playing with `panda-session` a little bit, it seems, irrespective of the HTTP response code of the iframe, the load event is fired, so this will shoot off as successful.

So:
* request -> `419`
* reauth -> `200`
* requestAttempt2 -> `404`
* `Session re-established`